.github/workflows/unit-tests.yml

```
@@ -22,39 +23,76 @@ jobs:
     strategy:
       matrix:
         include:
-          # MacOSX
-          - { compiler: clang,    os: macos-12,     buildType: Debug, cxxStandard: 14 }
-          # Linux GCC
-          # GCC 9 is known to show a few warnings that GCC 10 has "fixed", make sure this doesn't happen for us
-          - { compiler: gcc-9,    os: ubuntu-20.04, buildType: Debug }
-          - { compiler: gcc-10,   os: ubuntu-20.04, buildType: Debug, cxxStandard: 14, coverage: true }
-          - { compiler: gcc-10,   os: ubuntu-20.04, buildType: Debug, cxxStandard: 17 }
-          - { compiler: gcc-10,   os: ubuntu-20.04, buildType: Release, cxxStandard: 14 }
-          # Linux clang
-          - { compiler: clang-11, os: ubuntu-20.04, buildType: Debug, cxxStandard: 14, externalSanitizer: true }
-          - { compiler: clang-11, os: ubuntu-20.04, buildType: Debug, cxxStandard: 17 }
-          - { compiler: clang-11, os: ubuntu-20.04, buildType: Release, cxxStandard: 14 }
-          # Linux Latest GCC
-          - { compiler: gcc-12,   os: ubuntu-22.04, buildType: Debug, cxxStandard: 14, coverage: true }
-          # Linux Latest clang
-          - { compiler: clang-16, os: ubuntu-20.04, buildType: Debug, cxxStandard: 14, externalSanitizer: true }
+          # MacOSX:
+          #  Use system clang (14)
+          #   Use (compiler) default C++ standard (14)
+          #   Use system cmake (version gets ignored below)
+          #   Use default boost
+          - { compiler: clang,    os: macos-12,     type: Debug }
+
+          # Linux:
+          #  Oldest Compilers 
+          #   GCC 9 also known to show a few warnings that newer versions dont show
+          #   Use (compiler) default C++ standard (14)
+          #   Use system cmake
+          #   Use system boost (min version)
+          - { compiler: gcc-9,    os: ubuntu-20.04, type: Debug, cmake: 3.16.3, boost: 1.69.0 }
+          - { compiler: clang-10, os: ubuntu-20.04, type: Debug, cmake: 3.16.3, boost: 1.69.0, externalSanitizer: true }
+          #
+          #   Default compiler for Ubuntu 20.04
+          #   Use (compiler) default C++ standard (14)
+          #   Use default cmake
+          #   Use default boost
+          - { compiler: gcc-10,   os: ubuntu-20.04, type: Debug, coverage: true }
+          #
+          #   Default compilers for Ubuntu 22.04
+          #   Use C++ standard (17, default for gcc)
```

C&P made this hard to understand, maybe: 
```suggestion
          #   Use C++17 standard (default for gcc-11)
```

---

good point